### PR TITLE
Intoduction to redux

### PR DESCRIPTION
**Pull request by ifzyy**
### In this milestone

- Write actions and reducer for adding and removing books using the ducks pattern
- Added redux to my project

### CODE REVIEW

File: src/components/books.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ function Books() {
   return (
     <div className="mainBookContainer">
       <InputBooks />
+      <button type="button">Delete</button>
     </div>
```
Comment (mukhammadrizooff) [inline src/components/books.js:9]: - [ ] [REQUIRED] Good job so far! You forgot to create `Add` configuration as a physical button. Please fix it!

image

File: src/redux/Book/Book.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+const initialState = [];
+
+// Reducer
+const bookReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return [...state, action.book];
+    case REMOVE_BOOK:
+      return state.filter((book) => book.id !== action.id);
```
Comment (mukhammadrizooff) [inline src/redux/Book/Book.js:13]: - [ ] [REQUIRED] Your `Delete` button is not working, please fix it and test it!

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @ifzyy, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ifzyy** [comment]
Screenshot 2022-08-18 201620
The project does not require  delete button code implemented @mukhammadrizooff

File: src/redux/Book/Book.js (hunk 2 above)
Comment (mwanawabangona) [inline src/redux/Book/Book.js:13]: - [ ] `required`
- [ ] as mention by the previous reviewer please fix the `Delete Button` as the requirement

**Review by mwanawabangona — CHANGES_REQUESTED** [review]
Hi, @ifzyy ,

Great work on making the changes requested by a previous reviewer 👏

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lameck1 — COMMENTED** [review]
Hello @ifzyy 👋,

_This is an additional review due to the second opinion request._

Your implementation of the tasks for this milestone meets expectations. At this point, you are only expected to implement the redux logic. Therefore, you are right.

@mukhammadrizooff & @mwanawabangona, No changes in React components are required in this step. Please take note of this going forward.

As this was the only reason your project was not approved it will be approved as soon as possible.
Just hold on and don't ask for another review, you'll be able to submit this task on your dashboard ASAP.

Keep rocking 🚀

**Pull request merged**